### DISCUSSION TITLE
How can I trigger reaction automapping in a react component (e.g. via the Ketcher API)

### DISCUSSION THREAD

**Discussion by MartinHoffmannJena** [epam/ketcher#6018] [Q&A]
Hi,

I'm working on a react app using ketcher as the Cheminformatics sketching library. I would like to be able to trigger the "reaction auto mapping tool" by triggering a certain condition instead of having to click the button within the ketcher editor itself (see screenshot).

However I have not been able to find out how to do that, any help would be appreciated

![image](https://github.com/user-attachments/assets/0be4ac01-8412-48e4-a0ab-3e9f54eb0300)

**Comment by rrodionov91** [comment]
Hi @MartinHoffmannJena 

There is such way in ketcher api:
```
ketcher.editor.tool('reactionmap')
```

However it does not change active item in toolbar. It is a bug and need to be fixed, however it is not in our current plans unfortunately.
